### PR TITLE
Module 5 docker

### PR DESCRIPTION
**Pull request by TomJohnston73**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: poetry.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [virtualenvs]
-create = true
+create = false
```
Comment (TomJSoftwire) [inline poetry.toml:2]: This is no longer needed as docker handles py versioning etc

File: todo_app/start.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from app import create_app
```
Comment (TomJohnston73) [inline todo_app/start.py:1]: Still don't quite understand what these last 2 files do - found this online as a method for start gunicorn but don't really get the reasoning behind it. It does work though!

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+FROM python:3.11.0a6-slim-buster as base
+
+RUN apt-get update && apt-get install -y \
+    gcc \
+    libffi-dev
+
+RUN pip3 install 'poetry==1.1.11' \
+    && poetry config virtualenvs.create false
```
Comment (TomJohnston73) [inline Dockerfile:8]: Docker is already controlling the py env so we don't need this

File: todo_app/app.py [hunk 4]
```diff
@@ -1,9 +1,9 @@
 from flask import Flask, render_template, redirect
 from flask.globals import request
-from todo_app.data.trello_items import get_item, add_item, get_items, save_item
```
Comment (TomJohnston73) [inline todo_app/app.py:3]: These todo_app references weren't working in the container for some reason. Removing them seemed to fix the issue

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+WORKDIR /todo-app
+COPY poetry.lock .
+COPY poetry.toml .
+COPY pyproject.toml .
```
Comment (BenRamchandani) [inline Dockerfile]: You can combine these if you want
```suggestion
COPY poetry.lock poetry.toml pyproject.toml .
```

File: todo_app/start.py (hunk 2 above)
Comment (BenRamchandani) [reply todo_app/start.py:1]: It just provides an easy way to start the app, it's not necessary. Passing `wsgi:start` means find the start variable inside the wsgi module (i.e. wsgi.py) and use that as a Python WSGI compatible interface, which a Flask app is, so it works.
You could instead run `poetry run gunicorn 'app:create_app()' -b 0.0.0.0:80` (from inside your todo_app directory) and that would work fine too without these extra files.

File: .dockerignore [hunk 6]
```diff
@@ -0,0 +1 @@
+.*
```
Comment (BenRamchandani) [inline .dockerignore:1]: Would be good to exclude `**/__pycache__/` too

**Review by BenRamchandani — APPROVED** [review]
Looks great, the instructions in your README worked for me to get the app running in Docker. Go ahead and merge when you're ready.

**Pull request merged**